vision_agent/agent/execution.py

```diff
@@ -0,0 +1,287 @@
+"""This code is based off of code from CodeGeeX https://github.com/THUDM/CodeGeeX"""
+
+import contextlib
+import faulthandler
+import io
+import multiprocessing
+import os
+import platform
+import signal
+import tempfile
+import traceback
+import typing
+from pathlib import Path
+from typing import Dict, Generator, List, Optional, Union
+
+IMPORT_HELPER = """
+import math
+import re
+import sys
+import copy
+import datetime
+import itertools
+import collections
+import heapq
+import statistics
+import functools
+import hashlib
+import numpy
+import numpy as np
+import string
+from typing import *
+from collections import *
+from vision_agent.tools.tools_v2 import *
+"""
+
+
+def unsafe_execute(code: str, timeout: float, result: List) -> None:
+    with create_tempdir() as dir:
+        code_path = Path(dir) / "code.py"
+        with open(code_path, "w") as f:
+            f.write(code)
+
+        # These system calls are needed when cleaning up tempdir.
+        import os
+        import shutil
+
+        rmtree = shutil.rmtree
+        rmdir = os.rmdir
+        chdir = os.chdir
+
+        # Disable functionalities that can make destructive changes to the test.
+        reliability_guard()
+
+        try:
+            with swallow_io() as s:
+                with time_limit(timeout):
+                    # WARNING
+                    # This program exists to execute untrusted model-generated code. Although
+                    # it is highly unlikely that model-generated code will do something overtly
+                    # malicious in response to this test suite, model-generated code may act
+                    # destructively due to a lack of model capability or alignment.
+                    # Users are strongly encouraged to sandbox this evaluation suite so that it
+                    # does not perform destructive actions on their host or network.
+                    # Once you have read this disclaimer and taken appropriate precautions,
+                    # uncomment the following line and proceed at your own risk:
```

Comment:
Just making sure there is nothing to uncomment here ?

Reply:
Yeah this is from the original code base, I left a comment to link the original code at the top of the file. I will replace this file pretty soon with the notebook code execution